eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/primitive/ObjectBooleanHashMap.java

```diff
@@ -727,6 +727,20 @@ public boolean getIfAbsentPut(K key, boolean value)
         return value;
     }
 
+    @Override
+    public boolean getAndPut(K key, boolean putValue, boolean defaultValue)
+    {
+        int index = this.probe(key);
+        if (ObjectBooleanHashMap.isNonSentinel(this.keys[index]) && ObjectBooleanHashMap.nullSafeEquals(this.toNonSentinel(this.keys[index]), key))
+        {
+            boolean existingValue = this.values.get(index);
+            this.addKeyValueAtIndex(key, putValue, index);
```

Comment:
Can this be `this.values.set(index, putValue);`?

Reply:
Fixed.